app/src/contract_event_utils.py

```diff
@@ -1,7 +1,16 @@
+from time import time
+
 block_timestamp_cache = {}
 def block_timestamp(web3, block_number):
     global block_timestamp_cache
     if not isinstance(block_number, int) or block_number not in block_timestamp_cache:
-        block_timestamp_cache[block_number] = web3.eth.getBlock(block_number)["timestamp"]
+        block=web3.eth.getBlock(block_number)
+        if block!=None:
```

Comment:
Minor style nitpick: assignments and comparisons should have spaces around, e.g., `one != other`.